coding/using_boost.rst

```diff
@@ -4,24 +4,38 @@ Using Boost
 
 .. _cpp_using_boost:
 
-Certain Boost libraries are recommended; they should be used whenever applicable in preference to any other method of accomplishing the same effect. In general, any library that is tagged with *"Standard .... {something}"* in the `Boost library listing <http://www.boost.org/doc/libs>`_ falls into this category, unless the feature is already supported by the currently-mandated LSST ``g++`` version.
-Among others, this category includes:
+A Boost library shall be used in in preference to any other method of accomplishing the same effect only if:
 
-* ``current_function``
-* ``format``
-* ``regex`` (until we adopt ``gcc`` 4.9 as our minimum compiler version, at which point use ``std::regex``)
-* ``test``
+1. the effect cannot be accomplished with a C++11 standard language feature, and
+2. a C++ standard library equivalent is not available, or
+3. the C++ standard library equivalent is not usable with our minimum supported compiler version (i.e. ``gcc`` version 4.8).
```

Comment:
Precedence of "and" and "or" here is a little unclear :).  Adding "standard language feature" to point 3 might remove some ambiguity.